### PR TITLE
Add the release note for 1.16.0

### PR DESCRIPTION
**Pull request by trustin**
[Preview (PDF)](https://github.com/line/armeria/files/8512848/1.16.0.pdf)

### CODE REVIEW

File: site/src/pages/release-notes/1.16.0.mdx [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+  requests when `verboseResponses` option is enabled for easier debuggability. #4114 #4171
+- You can now specify query parameters additively by calling <type://WebClientRequestPreparation#queryParam(String,Object)>
+  #4173
+  ```java
+  // Send GET /api/v1/greet?foo=1&bar=2
+  WebClient
+    .of().prepare()
+    .get("/api/v1/greet")
+    .queryParam("foo", "1")
+    .queryParams("bar", "2") // 👈👈👈
```
Comment (ikhoon) [inline site/src/pages/release-notes/1.16.0.mdx:91]:
```suggestion
    .queryParams(Map.of("bar", "2")) // 👈👈👈
```

File: site/src/pages/release-notes/1.16.0.mdx [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+          .addService(...)
+          .build()
+      )
+    .build()
+  ```
+- You now get the `HeaderListSizeException` as expected when the length of request headers is too long. #4180
+- <type://EurekaEndpointGroup> now continues refreshing itself even if it failed to refresh once. #4206
+- <type://HealthCheckedEndpointGroup> now leaves a detailed error message if all endpoints are unhealthy
+  on the first check. #4200
+- Fixed a rare infinite loop in <type://WeightRampingUpStrategy> #TBD
```
Comment (ikhoon) [inline site/src/pages/release-notes/1.16.0.mdx:157]:
```suggestion
- Fixed a rare infinite loop in <type://EndpointSelectionStrategy#rampingUp()> #4219
```

File: site/src/pages/release-notes/1.16.0.mdx [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+  }
+  ```
+  - You can also specify a delimiter programmatically with <type://AnnotatedServiceBindingBuilder#queryDelimiter(String)>
+- You can now use the no-op `subscribe()` to drain elements from a <type://StreamMessage>  #4145 #4185
+  ```java
+  // The following statement prints 1, 2 and 3.
+  StreamMessage
+    .of(1, 2, 3)
+    .peek(value -> System.out.println(value))
+    .subscribe(); // 👈👈👈
```
Comment (ikhoon) [inline site/src/pages/release-notes/1.16.0.mdx:40]: Need `join()`?
```suggestion
    .subscribe().join(); // 👈👈👈
```

File: site/src/pages/release-notes/1.16.0.mdx (hunk 2 above)
Comment (trustin) [reply site/src/pages/release-notes/1.16.0.mdx]: Thanks!

File: site/src/pages/release-notes/1.16.0.mdx (hunk 3 above)
Comment (trustin) [reply site/src/pages/release-notes/1.16.0.mdx:40]: That's a good idea :+1:

File: site/src/pages/release-notes/1.16.0.mdx [hunk 4]
```diff
@@ -0,0 +1,233 @@
...
+    common_name=<the common name in TLS certificate>,
+    protocol=<h1 or h2>,
+    result=<success or failure>,
+    tls_protocol=<TLS protocol name, e.g. TLSv1.3>
+  }
+  ```
+- You can use the <type://@Delimiter> annotation in an annotated service to specify the delimiter of
+  a query parameter or header. #4060
+  ```java
+  public class MyAnnotatedService
```
Comment (ikhoon) [inline site/src/pages/release-notes/1.16.0.mdx:20]:
```suggestion
  public class MyAnnotatedService {
```

**Review by ikhoon — COMMENTED** [review]
Thank

**Review by ikhoon — APPROVED** [review]
🙇‍♂️🙇‍♂️🙇‍♂️

**Pull request merged**